File: core/LexicographicalNumbers/LexicographicalNumbers.java
package com.dbc;

import java.util.ArrayList;
import java.util.List;

public class LexicographicalNumbers {
    private final List<Integer> res = new ArrayList<>();
    private void order(int num, int n) {
        for (int i = 0; i < 10; i++) {
            if (num * 10 + i > n) return;
            res.add(num * 10 + i);
            order(num * 10 + i, n);
        }
    }

    public List<Integer> lexicalOrder(int n) {
        for (int i = 1; i < 10; i++) {
            if (i <= n) {
                res.add(i);
                order(i, n);
            }
        }
        return res;
    }
}

File: core/LexicographicalNumbers/LexicographicalNumbers.py
from typing import List


def lexicalOrder(n: int) -> List[int]:
    res = []

    def order(num):
        for i in range(10):
            if num * 10 + i > n:
                return
            res.append(num * 10 + i)
            order(num * 10 + i)

    for k in range(1, 10):
        if k > n:
            break
        res.append(k)
        order(k)

    return res


